mesonbuild/interpreter.py

```diff
@@ -2137,6 +2352,36 @@ def func_library(self, node, args, kwargs):
             return self.func_shared_lib(node, args, kwargs)
         return self.func_static_lib(node, args, kwargs)
 
+    def func_component(self, node, args, kwargs):
+        if len(args) != 1:
+            raise InterpreterException('Component takes exactly one argument',
+                                       'its name, not "%s"' % args)
+
+        name = args[0]
+        deps = mesonlib.flatten(kwargs.get('dependencies', []))
+        default_value = kwargs.get( 'default',
+                                   self.coredata.builtins['components_default_value'].value)
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->